x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextModelSettings.java

```diff
@@ -98,4 +138,46 @@ public Integer dimensions() {
     public SimilarityMeasure similarity() {
         return similarity;
     }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(TASK_TYPE_FIELD.getPreferredName(), taskType.toString());
+        builder.field(INFERENCE_ID_FIELD.getPreferredName(), inferenceId);
+        if (dimensions != null) {
+            builder.field(DIMENSIONS_FIELD.getPreferredName(), dimensions);
+        }
+        if (similarity != null) {
+            builder.field(SIMILARITY_FIELD.getPreferredName(), similarity);
+        }
+        return builder.endObject();
+    }
+
+    public void validate() {
+        switch (taskType) {
+            case TEXT_EMBEDDING:
+            case SPARSE_EMBEDDING:
+                break;
+
+            default:
+                throw new IllegalArgumentException("Wrong [" + TASK_TYPE_FIELD.getPreferredName() + "], expected " +
+                        TEXT_EMBEDDING + "or " + SPARSE_EMBEDDING + ", got " + taskType.name());
+        }
```

Comment:
Doesn't a valid model settings also require an inference ID?

Reply:
It's already checked in the constructor, inferenceId cannot be null.

Reply:
Got it, didn't see it initially because that code didn't change

Reply:
Nitpick: Since this method is only about validating the task type, what if we:

- Renamed it to `validateTaskType`
- Used it to check if `taskType` is null
- Called it from `SemanticTextModelSettings(TaskType taskType, String inferenceId, Integer dimensions, SimilarityMeasure similarity)` so we always check if the task type is valid

Reply:
That's the intent. The validate call was put on the wrong constructor so I fixed that but taskType is already required to be non-null. I also added more validations so I think the naming is fine.